OpenRA.Game/Graphics/TerrainSpriteLayer.cs

```diff
@@ -32,7 +32,7 @@ public sealed class TerrainSpriteLayer : IDisposable
 		readonly bool restrictToBounds;
 
 		readonly WorldRenderer worldRenderer;
-		readonly Map map;
+		readonly IMap map;
```

Comment:
The `TerrainSpriteLayer` constructor should be checking that `world.Map` implements `IMapTiles` and throw an exception if it doesn't (see how the tileset code does this for examples).